ballerina/swiftmt_parser.bal

```diff
@@ -0,0 +1,272 @@
+    xml customizedXml = check customizeGeneratedXml(generatedXml);
+    string messageType = (customizedXml/**/<messageType>).data();
+    string validationFlag = (customizedXml/**/<ValidationFlag>/<value>).data();
+    typedesc<record {}>? messageRecord;
+
+    if validationFlag.equalsIgnoreCaseAscii("STP") || validationFlag.equalsIgnoreCaseAscii("REMIT") || validationFlag.equalsIgnoreCaseAscii("COV") {
+        messageRecord = messageMapper[messageType + validationFlag];
+        if messageRecord is null {
+            return error("SWIFT message type is invalid or not supported.");
+        }
+        return xmldata:parseAsType(customizedXml, t = messageRecord);
+    }
+    if check int:fromString(messageType.substring(1, 3)) >= 90 {
+        messageRecord = messageMapper[messageType.substring(1, 3)];
+        if messageRecord is null {
+            return error("SWIFT message type is invalid or not supported.");
+        }
+        return xmldata:parseAsType(customizedXml, t = messageRecord);
+    }
+    messageRecord = messageMapper[messageType];
+    if messageRecord is null {
+        return error("SWIFT message type is invalid or not supported.");
+    }
+    return xmldata:parseAsType(customizedXml, t = messageRecord);
+}
+
+# Customizes a generated XML by renaming specific tags and components.
+#
+# + customXml - The input XML object that represents the SWIFT message. This XML is generated from the original SWIFT message.
+# + return - Returns the customized XML with renamed tags and components. If a transaction sequence is required, it is added to the message. In case of an error, returns an error indicating the issue.
+isolated function customizeGeneratedXml(xml customXml) returns xml|error {
+    boolean isTransactionSequencerequired = false;
+    string messageType = (customXml/**/<messageType>).data();
+    boolean isNotCommonMessageType = check int:fromString(messageType.substring(1, 3)) < 90;
+
+    foreach xml tagElement in (customXml/**/<block3>).elementChildren() {
+        if (tagElement/<name>).data().equalsIgnoreCaseAscii("121") {
+            tagElement.setName("NdToNdTxRef");
+        } else if (tagElement/<name>).data().equalsIgnoreCaseAscii("119") {
+            tagElement.setName("ValidationFlag");
+        }
+    }
+
+    string[][]? csvContent1 = FIELDNAMESPEC[messageType.substring(0, 1)];
+    if csvContent1 is null {
+        return error("SWIFT message type is invalid or not supported.");
+    }
+    foreach xml tagElement in (customXml/**/<block4>).elementChildren() {
+        string name = (tagElement/<name>).data();
+        if name.equalsIgnoreCaseAscii("21") {
+            isTransactionSequencerequired = true;
+        }
+        tagElement.setName("MT".concat(name));
+        string componentName = "";
+        foreach xml component in tagElement.elementChildren() {
+            if component.getName().equalsIgnoreCaseAscii("component") {
+                foreach string[] line in csvContent1 {
```

Comment:
You can use `ballerina/data.csv` for csv processing, so the logic can be much simpler.

Reply:
Sorry, previously the csv was used but now it has been changed to array. So I will rename the variable.

Reply:
fixed in `99b8222`